app/views/SelectedUsersView.tsx

```diff
@@ -1,64 +1,53 @@
-import React from 'react';
-import { StackNavigationProp } from '@react-navigation/stack';
+import { Q } from '@nozbe/watermelondb';
 import { RouteProp } from '@react-navigation/native';
+import orderBy from 'lodash/orderBy';
+import React from 'react';
 import { FlatList, View } from 'react-native';
 import { connect } from 'react-redux';
-import orderBy from 'lodash/orderBy';
-import { Q } from '@nozbe/watermelondb';
 import { Subscription } from 'rxjs';
 
+import { addUser, removeUser, reset } from '../actions/selectedUsers';
+import { themes } from '../constants/colors';
+import * as HeaderButton from '../containers/HeaderButton';
 import * as List from '../containers/List';
-import database from '../lib/database';
-import RocketChat from '../lib/rocketchat';
-import UserItem from '../presentation/UserItem';
 import Loading from '../containers/Loading';
-import I18n from '../i18n';
-import log, { events, logEvent } from '../utils/log';
+import SafeAreaView from '../containers/SafeAreaView';
 import SearchBox from '../containers/SearchBox';
-import * as HeaderButton from '../containers/HeaderButton';
 import StatusBar from '../containers/StatusBar';
-import { themes } from '../constants/colors';
-import { withTheme } from '../theme';
+import { IApplicationState, IBaseScreen } from '../definitions';
+import I18n from '../i18n';
+import database from '../lib/database';
+import RocketChat from '../lib/rocketchat';
+import UserItem from '../presentation/UserItem';
+import { ISelectedUser } from '../reducers/selectedUsers';
 import { getUserSelector } from '../selectors/login';
-import { addUser as addUserAction, removeUser as removeUserAction, reset as resetAction } from '../actions/selectedUsers';
+import { ChatsStackParamList } from '../stacks/types';
+import { withTheme } from '../theme';
 import { showErrorAlert } from '../utils/info';
-import SafeAreaView from '../containers/SafeAreaView';
+import log, { events, logEvent } from '../utils/log';
 import sharedStyles from './Styles';
-import { ChatsStackParamList } from '../stacks/types';
 
 const ITEM_WIDTH = 250;
 const getItemLayout = (_: any, index: number) => ({ length: ITEM_WIDTH, offset: ITEM_WIDTH * index, index });
 
-interface IUser {
-	_id: string;
-	name: string;
-	fname: string;
-	search?: boolean;
-	// username is used when is from searching
-	username?: string;
-}
 interface ISelectedUsersViewState {
 	maxUsers?: number;
-	search: IUser[];
-	chats: IUser[];
+	search: ISelectedUser[];
+	chats: ISelectedUser[];
 }
 
-interface ISelectedUsersViewProps {
-	navigation: StackNavigationProp<ChatsStackParamList, 'SelectedUsersView'>;
+interface ISelectedUsersViewProps extends IBaseScreen {
```

Comment:
the idea from typing the navigation with a generic type, is still using the react navigation typed already implemented. 
So with that changed, we need pass the props to the `IBaseScreen`. 
`IBaseScreen<ChatsStackParamList, 'SelectedUsersView'>` like this.
With this, we still have the types when we navigate to the screens... 

Before:
![Screen Shot 2021-12-29 at 15 21 32](https://user-images.githubusercontent.com/14894380/147692110-3714196f-890f-4358-ab45-2ba48c0bab05.png)

After:
![Screen Shot 2021-12-29 at 15 22 10](https://user-images.githubusercontent.com/14894380/147692140-1dd6d573-dbf7-439b-916a-90761373210e.png)